lazy_tensor_core/lazy_tensor_core/csrc/tensor_util.h

```diff
@@ -32,24 +32,6 @@ std::vector<compiler::BackendDataPtr> CreateTensorsData(
     const std::vector<at::Tensor>& tensors,
     const std::vector<Device>& devices);
 
-template<typename... TupleType>
-std::vector<std::vector<int64_t>> CreateComputationShapeFromMetaTensors(const std::tuple<TupleType...>& tensors) {
```

Comment:
There is a C++14 equivalent technique to do the same work. Just not as concise as this one. If you think code-gen is better, I'm fine with that. Otherwise, I can do a quick rewrite of the helpers.

Reply:
nah lets just leave it codegen. not worth the effort imo.